system-config/systemd-cryptsetup@swap.service

```diff
@@ -0,0 +1,35 @@
+[Unit]
+Description=Qubes OS Encrypted Swap
+
+# This is a dependency of sysinit.target, so it cannot depend on it (deadlock).
+DefaultDependencies=no
+
+# Do not stop this when trying to isolate a unit.
+IgnoreOnIsolate=true
+
+# Default dependencies for any encrypted volume.
+After=cryptsetup-pre.target systemd-udevd-kernel.socket systemd-random-seed.service
+
+# Ensure that this unit comes after its backing device,
+# and is shut down if its backing device is shut down.
+After=dev-qubes_dom0-swap.device
+BindsTo=dev-qubes_dom0-swap.device
```

Comment:
Even in LVM setup, the VG name can be different, and LV name can be different too...
Previous version also had this issue, but now it's all over the place instead of just StartExec line.

Maybe it's possible to rely on generators (and maybe the default ones will be enough)? Like, setup backing device via /etc/crypttab, and then use /dev/mapper/swap as the actual swap (via /etc/fstab).

Reply:
That’s absolutely fine, and it’s what I do on my local system.  I have this line in `/etc/crypttab`:

```crypttab
swap /dev/qubes_dom0/swap /dev/urandom plain,swap,cipher=aes-xts-plain64
```

And this in `/etc/fstab` (modulo padding whitespace):

```fstab
/dev/mapper/swap none swap defaults 0 0
```

The reason that this PR does not take that approach is that these files are generated by Anaconda, which I have never touched before.  I can try, though.